src/Screen/Screen.php

```diff
@@ -266,11 +266,18 @@ protected function fillPublicProperty(Repository $repository): void
      */
     protected function getPublicPropertyNames(): Collection
     {
-        $reflections = (new \ReflectionClass($this))->getProperties(\ReflectionProperty::IS_PUBLIC);
-
-        return collect($reflections)
-            ->filter(fn (\ReflectionProperty $property) => ! $property->isStatic())
-            ->map(fn (\ReflectionProperty $property) => $property->getName());
+        $reflection = (new \ReflectionClass($this));
+
+        $properties = new Collection();
```

Comment:
Do we really need to iterate through all parent classes to find properties?

It seems to me that the following call would yield the same result:

```php
ReflectionClass(static::class)
```

Or am I mistaken?

Reply:
@tabuna You are right, your code will enumerate all public properties without looping.

On the other side it will not protect public properties defined before the original Screen class if not redefined in a sub class but at the moment I dont see properties defined in base classes.

If you agree I will update the PR with your optimized code

Reply:
I reviewed it, and indeed, we are using the public property `$middleware` because we inherit from the controller. I think we can handle the exception in the same way as in the `getAvailableMethods` method. Could you please update the PR?

Reply:
Hi @tabuna,
I made a rebase with latest changes and improved the code with all the suggestions.